### PR TITLE
Use *either* addon or app options, never both.

### LINKED ISSUE #204: Bug: incorrect options handling causes broccoli-funnel error
When using the `passthrough` option we [added] in [v7.2.0], the addon crashes with [a broccoli-funnel error][funnel], because the options are incorrectly merged for both the app and the addon. You can verify this by setting any `srcDir` in the `passthrough` options for an app when using ember-cli-sass in both an app *and* an addon, and both are on at least 7.2.0. (I discovered this when attempting to upgrade ember-freestyle to use ember-cli-sass@7.2.0.)

[added]: https://github.com/aexmachina/ember-cli-sass/commit/9a65013bfaa06732e291235e0f638414bef3c06c
[v7.2.0]: https://github.com/aexmachina/ember-cli-sass/compare/v7.1.4...v7.2.0
[funnel]: https://github.com/broccolijs/broccoli-funnel/issues/105

This is caused by merging the options from the parent *app* when in an addon:

https://github.com/aexmachina/ember-cli-sass/blob/9de7fe427729e74d1d086aad6446e9e88a222a39/index.js#L59

Per a conversation with @rwjblue, we only want to use the `app` options when `parent` *is* the app, rather than an addon, i.e. when `this.project == this.parent`. The solution is to adopt [the model used by ember-cli-babel][babel]: never to use *both* the app and addon options; instead use addon options if available, or app options if not.

[babel]: https://github.com/babel/ember-cli-babel/blob/e31c79efc48b007e2b4f61356cbf320d954a2134/index.js#L208-L210

I expect to have a PR up for this tomorrow!

### PR DESCRIPTION
**Pull request by chriskrycho**
Fixes #204.

### CODE REVIEW

**Comment by chriskrycho** [comment]
Would love eyes here from someone like @stefanpenner, @rwjblue, @dfreeman, etc.

File: index.js [hunk 1]
```diff
@@ -72,11 +72,15 @@ module.exports = {
...
     var env  = process.env.EMBER_ENV;
-    var options = (this.app && this.app.options && this.app.options.sassOptions) || {};
-    var parentOption = (this.parent && this.parent.app && this.parent.app.options && this.parent.app.options.sassOptions) || {};
     var envConfig = this.project.config(env).sassOptions;
 
-    Object.assign(options, parentOption);
+    // *Either* use the options for an addon which is consuming this, *or* for
+    // an app which is consuming this, but never *both* at the same time. The
+    // special case here is `"dummy"`, which only applies when testing an addon.
+    var useApp = this.app.name === 'dummy' || this.app.name === this.project.name();
```
Comment (rwjblue) [inline index.js]: I think I would use `this.isDevelopingAddon()` here instead of the manual checks for `dummy`...

File: index.js [hunk 2]
```diff
@@ -72,11 +72,15 @@ module.exports = {
...
+    var options = useApp
+      ? (this.app && this.app.options && this.app.options.sassOptions) || {}
```
Comment (rwjblue) [inline index.js]: Any reason not to do:

```js
var options = (
  (this.parent && this.parent.options && this.parent.options.sassOptions) ||
  (this.app && this.app.options && this.app.options.sassOptions) ||
  {}
);
```

That might allow you to remove the usage of `useApp`...

File: index.js (hunk 1 above)
Comment (chriskrycho) [reply index.js]: Ah, perfect.

File: index.js (hunk 2 above)
Comment (chriskrycho) [reply index.js]: That makes some sense and is worth testing – though, unfortunately, my test environment where I could reliably repro the bug behind this is now unavailable; I'll have to see if I can set up a smaller repro of the original problem.

**Comment by chriskrycho** [comment]
Updated with the suggested changes, but I need to whip up a small demo app to confirm that this does what I expect it too.

**Comment by chriskrycho** [comment]
Confirmed via [this example code][e.g.] that the updated version still resolves this. Good for merge.

[e.g.]: https://github.com/chriskrycho/sass-fix-test

**Comment by aexmachina** [comment]
Thanks! Pushed v10.

I do intend to transfer ownership, but don't have bandwidth right now.

**Comment by chriskrycho** [comment]
Thanks for getting this merged and published, @aexmachina!

**Pull request merged**